src/mlpack/methods/lsh/lsh_search.hpp

```diff
@@ -83,15 +83,19 @@ class LSHSearch
   ~LSHSearch();
 
   /**
-   * Train the LSH model on the given dataset.  This means building new hash
-   * tables.
+   * Train the LSH model on the given dataset.  If a correct vector is not
+   * provided, this means building new hash tables. Otherwise, we use the ones
+   * provided by the user.
    */
   void Train(const arma::mat& referenceSet,
              const size_t numProj,
              const size_t numTables,
              const double hashWidth = 0.0,
              const size_t secondHashSize = 99901,
-             const size_t bucketSize = 500);
+             const size_t bucketSize = 500,
+             const std::vector<arma::mat> &projection
+             = std::vector<arma::mat>()
```

Comment:
Do you think we could use `arma::cube` here instead?  We know how many projection tables we have and how many projections are in each table (and it's the same number of projections per table), so we don't need the extra overhead of the vector.  I know that's not code you wrote, but I wanted to ask if you think we could make that change (I'll do it if you think it's reasonable, unless you beat me to it :)).